src/components/AgenciesPage/reducers/agenciesReducer.ts

```diff
@@ -0,0 +1,35 @@
+import { AgenciesActionTypes } from '../actions/ActionTypesAgencies';
+import type { AgenciesState, Agency, AgenciesAction } from '../types/Agencies';
+
+const initialState: AgenciesState = {
+  agencies: [],
+  isLoading: false,
+  error: '',
+};
+
+export const agenciesReducer = (
+  state = initialState,
+  { type, payload }: AgenciesAction,
+) => {
+  switch (type) {
+    case AgenciesActionTypes.FETCH_AGENCIES_REQUEST:
+      return {
+        ...state,
+        isLoading: true,
+      };
+    case AgenciesActionTypes.FETCH_AGENCIES_SUCCESS:
+      return {
+        agencies: payload as Agency[],
+        isLoading: false,
+        error: '',
+      };
+    case AgenciesActionTypes.FETCH_AGENCIES_FAILURE:
+      return {
+        ...state,
+        isLoading: false,
+        error: payload as string,
```

Comment:
why as?

Reply:
because the type can be different, because  we have different types of our state values:  string | boolean | Agency[]

Reply:
But you defined that your payload can be string | boolean etc.

Reply:
yes 
because 
 `interface AgenciesState {
  agencies: Agency[];
  isLoading: boolean;
  error: string;
}`
` interface AgenciesAction {
  type: AgenciesActionTypes;
  payload?: AgenciesState[keyof AgenciesState];       
}`    

error is a string type 
but payload can be different type

Reply:
Nope, payload can't be different.  You have 3 different interfaces for success, failure etc.

Reply:
fix as Dima, change interface on more more detailed type AgenciesAction